app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            total = self.km + other.km
+        elif isinstance(other, (int, float)):
+            total = self.km + other
+        return Distance(total)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        if isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: Distance | int | float) -> Distance:
```

Comment:
Also `other` can not be of type `Distance` in this method, because if you multiply km * km you get square km